src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1536,6 +1526,37 @@ private function parse_next_tag() {
 				return true;
 			}
 
+			if ( $at > $was_at ) {
+				/*
+				 * A "<" has been found in the document. That may be the start of another node, or
+				 * it may be an "ivalid-first-character-of-tag-name" error. If this is not the start
+				 * of another node the "<" should be included in this text node and another
+				 * termination point should be found for the text node.
+				 *
+				 * @see https://html.spec.whatwg.org/#tag-open-state
+				 */
+				if ( strlen( $html ) > $at + 1 ) {
+					$next_character  = $html[ $at + 1 ];
+					$at_another_node =
+						'!' === $next_character ||
+						'/' === $next_character ||
+						'?' === $next_character ||
+						( 'A' <= $next_character && $next_character <= 'z' );
```

Comment:
I don't love this, I think it's doing character code comparison but I couldn't find conclusive docs.

I considered [`ctype_alpha`](https://www.php.net/manual/en/function.ctype-alpha.php) but apparently that's locale dependent so best avoided. We want ascii alpha.

Reply:
based on the PHP docs for the comparison operators I think this is valid as-is, as long as we're comparing strings to strings